### PR TITLE
optimizations | autocomplete is faster/lighter on cpu/network #225

### PR DESCRIPTION
**Pull request by sriv**
cache completion sets per project

### CODE REVIEW

**Comment by ocary** [comment]
@sriv - I'm excited to try this out once it gets merged and a nightly is available.   Thank you for working on this.

File: Gauge.VisualStudio/SolutionsEventListener.cs [hunk 1]
```diff
@@ -77,6 +80,8 @@ public int OnBeforeCloseProject(IVsHierarchy pHierarchy, int fRemoved)
             var project = pHierarchy.ToProject();
             var slugifiedName = project.SlugifiedName();
 
+            StatusBarLogger.Log($"Initializing Gauge Project Cache: {project.Name}");
```
Comment (nehashri) [inline Gauge.VisualStudio/SolutionsEventListener.cs]: Gauge project cache is being deleted here

File: Gauge.VisualStudio/UI/ClassPicker.xaml.cs [hunk 2]
```diff
@@ -20,6 +20,7 @@
 using System.Windows.Media;
 using EnvDTE;
 using Microsoft.CSharp;
+using Gauge.VisualStudio.Model;
```
Comment (nehashri) [inline Gauge.VisualStudio/UI/ClassPicker.xaml.cs]: unused import.

File: Gauge.VisualStudio.Model/IProject.cs [hunk 3]
```diff
@@ -27,5 +28,8 @@ public interface IProject
         bool HasDuplicateImplementation(ITextSnapshotLine line);
         IEnumerable<string> GetAllStepsForCurrentProject();
```
Comment (nehashri) [inline Gauge.VisualStudio.Model/IProject.cs]: Is this still used?

**Pull request merged**